testing/server-item-renderer-with-debug-ui.tsx

```diff
@@ -58,7 +67,16 @@ export const ServerItemRendererWithDebugUI = ({
         );
     };
 
-    const renderedQuestion: PerseusRenderer = answerless
+    // `reviewMode` and `showSolutions` require answerful data by definition,
+    // so only use answerless data if those are not enabled. Also updates the
+    // startAnswerless toggle so it actually switches between answerless and answerful.
+    const shouldUseAnswerless =
+        useAnswerless &&
+        !reviewMode &&
+        (showSolutions === "none" || !showSolutions) &&
+        startAnswerless;
+
+    const renderedQuestion: PerseusRenderer = shouldUseAnswerless
```

Comment:
```suggestion
    // `reviewMode` and `showSolutions` require answerful data by definition,
    // so only use answerless data if those are not enabled.
    const shouldUseAnswerless =
        useAnswerless &&
        !reviewMode &&
        (showSolutions === "none" || !showSolutions);

    const renderedQuestion: PerseusRenderer = shouldUseAnswerless
```

You don't want to directly use `startAnswerless` because `useAnswerless` is the stateful version of `startAnswerless`. This also doesn't update `startAnswerless` so the comment is a little confusing.

Just realizing `useAnswerless` should probably be renamed so it's not confused with a hook 🤦 That was my bad. Maybe just `answerless`/`setAnswerless` or something. This is getting a little confusing, probably due to bad design decisions on my part...but it's test code, so we can always fix it later.

Reply:
@handeyeco Okay, I renamed `startAnswerless` to `answerless/setAnswerless`. 

Regarding `startAnswerless` being there (on L77 of the highlighted code), when it is there, toggling `startAnswerless` in the Controls for answerless stories immediately changes the data from answerless to answerful. It actually acts as a functioning toggle. It doesn't seem to work if `startAnswerless` starts out being false though. Those need a page refresh. Should I still take `startAnswerless` out from here though?

Reply:
I'll just remove the `startAnswerless` from the conditional for now, but it can be added back or refactored if we want the toggle to work.